templates/express_oauth_microsoft/router/auth.js

```diff
@@ -0,0 +1,39 @@
+import express from 'express'
+import passport from 'passport'
+import jwt from 'jsonwebtoken'
+
+const router = express.Router()
+
+// token is verified if exists otherwise redirects to microsoft auth
+router.get('/login', (req, res) => {
+    const token = req.cookies?.token
+    if (token) {
+        try {
+            const user = jwt.verify(token, process.env.JWT_SECRET)
+            return res.redirect('/')
+        } catch {
+            res.clearCookie('token')
+        }
+    }
+    res.redirect('/auth/microsoft/redirect')
```

Comment:
Enclose in a try catch block. Catch will throw 500 and write to a log file. Refer the `express_mysql` template for doing it.